EIPS/eip-7791.md

```diff
@@ -90,8 +103,8 @@ The proposed cost of this opcode is identical to the recently proposed `PAY` opc
 - Charging gas instead of ETH: Charging ETH directly complicates the user experience and prevents contract authors from participating in fluctuations in gas demand directly.
 - For the value of `gas_price`, use the same gas price as which is used to calculate the tx cost in ETH. This leads to the most consistent computation between `GAS2ETH` and the user's experience when creating a transaction.
 - Separate budget: Having a separate budget prevents certain types of DoS attacks where usage of `GAS2ETH` can interfere with the actual budget needed for execution of the contract. (For example, a target contract can "waste" gas with `GAS2ETH`, resulting in the caller not having enough gas to finish execution). This also results in `GAS2ETH` usage not affecting the burn rate. The intuition is that, since it doesn't actually use computational resources (which gas normally "accounts" for), it shouldn't affect blockspace computations or the base fee.
-- Limit being 100% of the gas limit: 100% of the gas limit is a "natural" bound on the gas used and provides some protection to the user from unbounded `GAS2ETH` spending. In the future, a new EIP-2718-based transaction type could be added which includes an explicitly user-specified `GAS2ETH` limit.
-- No exceptional halt when `gas_amount` exceeds the remaining `GAS2ETH` budget: since the `GAS2ETH` budget is not introspectable, the contract would otherwise not have a way to gracefully avoid an exceptional halt.
+- Budget being 100% of the gas limit: 100% of the gas limit is a "natural" bound on the gas used and provides some protection to the user from unbounded `GAS2ETH` spending. In the future, a new [EIP-2718](./eip-2718.md)-based transaction type could be added which includes an explicitly user-specified `GAS2ETH` budget.
+- No exceptional halt when `gas_amount` exceeds the remaining budgets, both the current call frame's local `GAS2ETH` allowance and the transaction-global `GAS2ETH` budget. Since these budgets are not introspectable, the contract would otherwise have no way to gracefully avoid an exceptional halt.
```

Comment:
i think we should add a bullet point to the rationale here:

> Bounding by execution gas ensures that untrusted callees cannot consume more GAS2ETH than the gas explicitly forwarded to them

Reply:
addressed via https://github.com/ethereum/EIPs/pull/10551/commits/a3bff7c50d85340a85263b309405e7ffadbda317.